doc/howto/write_an_alphabet/index.md

```diff
@@ -0,0 +1,149 @@
+In this section we want to implement a seqan3::Alphabet that consists of the characters `S` and `W` to represent
+strong and weak nucleobases.
+
+Let's start with a simple struct that only holds the alphabet's numerical representation, namely the **rank** value:
+\snippet dna2_only_rank.cpp struct
+
+If you want SeqAn's algorithms to accept it as an alphabet, you need to make sure that your type
+satisfies the requirements of the seqan3::Alphabet concept. A quick check can reveal that this is not the case:
+\snippet dna2_only_rank.cpp alphabet_concept
+
+\note
+A `static_assert()` will only tell you whether the expression is true or not.
+If you want the compiler to tell you **why** the concept is not modelled,
+you can construct a dummy requirement like this:
+\snippet dna2_alphabet.cpp dummy_requirement
+This will fail with a (slightly long and need-to-get-used-to) error message telling you
+that `foo()` cannot be called with `dna2` because `constraints are not satisfied ... `.
+
+## Prerequisites
+
+A look at the documentation of seqan3::Alphabet will reveal that it is actually a refinement of other concepts,
+more precisely seqan3::Semialphabet which in turn refines std::CopyConstructible and std::StrictTotallyOrdered.
+Let's check those:
+\snippet dna2_only_rank.cpp other_concepts
+
+You should see that your type models only the std::CopyConstructible concept.
+Let's have a look at the documentation for std::StrictTotallyOrdered. Can you guess what it describes?
+
+It describes the requirements for types that are comparable via `<`, `<=`, `>` and `>=`.
+This is useful so that you can sort a text over the alphabet, for example.
+Additionally, std::StrictTotallyOrdered is again a refinement of std::EqualityComparable,
+which requires the `==` and `!=` operators, so let's first implement those with free member functions.
+\snippet dna2_semialphabet.cpp equality
+
+\assignment{Excercise}
+Implement the inequality operator (!=) for `dna2`.
+\endassignment
+\solution
+\snippet dna2_semialphabet.cpp inequality
+\endsolution
+
+We see that our type now models std::EqualityComparable, which is a prerequisite of std::StrictTotallyOrdered.
+\snippet dna2_comparison_operators.cpp equality_comparable_concept
+
+\assignment{Excercise}
+Implement the four comparison operators *as free member functions* and verify that your type models
+std::StrictTotallyOrdered.
+\endassignment
+\solution
+\snippet dna2_comparison_operators.cpp comparison
+\endsolution
+
+## Semialphabet
+
+Let's move on to the more interesting concepts. seqan3::Semialphabet requires the *rank interface*
+that we have introduced in the [alphabet tutorial](\ref tutorial_alphabets). If we define the member functions
+`to_rank`, `assign_rank` and `alphabet_size` for our type, SeqAn automatically exposes them to the respective global
+functions `seqan3::to_rank`, `seqan3::assign_rank_to` and `seqan3::alphabet_size`.
+Note that `assign_rank` is only required for seqan3::WritableSemialphabet.
```

Comment:
This seems a little outdated and should address the CPOs more closely. Maybe something like this:

```
Let's move on to the more interesting concepts. seqan3::Semialphabet constitutes the *rank interface*
that we introduced in the [alphabet tutorial](\ref tutorial_alphabets). Have a look at the API reference again. Beyond the conceptional requirements, it also requires that seqan3::alphabet_size and seqan3::to_rank can be called on your alphabet.

There are different ways to satisfy seqan3::alphabet_size and seqan3::to_rank, have a look at the respective API reference and also [the documentation on customisation points](link).

In this case we choose to implement the functionality as member functions:
```